### DISCUSSION TITLE
How to add organicmaps api to an app in Android Studio

### DISCUSSION THREAD

**Discussion by prepbgg2** [organicmaps/organicmaps#8059] [Q&A]
I am a part time programmer who has developed a couple of apps for personal use. I have successfully linked to Maps.Me in the past and would now like to link to Organic Maps. Therefore, I would like to add the organicmaps API Library to my project in Android Studio. The [API Readme](https://github.com/organicmaps/api-android/?tab=readme-ov-file#readme) says I need to be familiar with the concept of "library projects" but the link that follows links to an Android Developer page which seems to be about developing a library. Can anyone provide a link to information on how to add an existing library (such as the lib in the repository that I have downloaded) to my project, please?

**Comment by biodranik** [comment]
The referenced project is a library. You can either embed it into your project, or open URLs/intents directly, examples are at https://omaps.app/api

**Reply by prepbgg2** [reply]
Thanks for the reply. From the examples it looks as if coding will be easier if I can embed the library. My question is how do I do that? When I search on line I can't find anything from the Android Developers website. Some websites say I should reference the library in an online repository. Others say I should import a jar or aar package but I don't see either of them in the "Organic Maps api-android-master" download. Exactly how do I use the "lib" folder in the download?

**Reply by prepbgg2** [reply]
I should have added that all I want to do in my app is include a call like `Api.showPointOnMap(this, lat, lon, name);` as shown in the "simplest usage" example on [API ReadMe](https://github.com/organicmaps/api-android/?tab=readme-ov-file#readme)

**Reply by biodranik** [reply]
If you don't know how to add a library or copy the code, you can write it yourself, by programmatically forming and opening URL like
`om://map?v=1&ll=22.17319,-159.65687&n=Kalalau%20Camping`

**Comment by prepbgg2** [comment]
I really would like to add the library to my app if I can.

I've tried (in Android Studio) going to File>New>Import Module and in the "Import module from source" window pointed the "Source directory" to the "api-android-master\lib" folder which I extracted from the "Organic Maps api-android-master.zip" download.  I then named the lib/module "OrganicMaps".

After doing that Gradle ran but failed with the following messages:

> A problem occurred evaluating project ':OrganicMaps'.
> > Could not get unknown property 'targetSdkVersion' for project ':OrganicMaps' of type org.gradle.api.Project.
> * Exception is:
> org.gradle.api.GradleScriptException: A problem occurred evaluating project ':OrganicMaps'.
> 	at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:93)
>         ....

 
Can you tell me where I've gone wrong, please?

**Reply by prepbgg2** [reply]
I'm looking at your suggestion
> If you don't know how to add a library or copy the code, you can write it yourself, by programmatically forming and opening URL like
om://map?v=1&ll=22.17319,-159.65687&n=Kalalau%20Camping

As I've mentioned, I would like to add the library. I hope someone can help me.

Failing that, I think I would be able to copy the code. That's what I did for Maps.Me and the Organic Maps code looks very similar.

However, your suggestion of forming and opening a URI looks interesting. It presumably would be simpler than copying the API code. I'm sure I would be able to form the URI. Can you point me to an example showing the code needed to open it, please?

**Reply by biodranik** [reply]
Organic Maps should already open such URLs, the parsing code is in the main repository.

**Reply by prepbgg2** [reply]
Thanks again for your reply.

In fact, it seems that I had imported the library correctly. All I needed to do in addition was to click the "Sync Project with Gradle files" button near the top right of the Android Studio window! (I had assumed that when Gradle ran automatically after I imported the library it would do all the syncing necessary. It seems this assumption was wrong!) I've now clicked the button and Gradle says "BUILD SUCCESSFUL".
